simple_server/Message.cpp

```diff
@@ -0,0 +1,126 @@
+#include "../inc/Message.hpp"
+
+Message::Message(void) : _src(""), _nick(""), _user(""), _hostname(""), _verb("") {}
+
+Message::~Message(void) {}
+
+// ------- GETTER AND SETTER -------
+
+std::string	Message::getSrc(void) const
+{
+	return (this->_src);
+}
+
+std::string Message::getNick(void) const
+{
+	return (this->_nick);
+}
+
+std::string	Message::getUser(void) const
+{
+	return (this->_user);
+}
+
+std::string	Message::getHostname(void) const
+{
+	return (this->_hostname);
+}
+
+std::string	Message::getVerb(void) const
+{
+	return (this->_verb);
+}
+
+std::vector<std::string>	Message::getParameters(void) const
+{
+	return (this->_parameters);
+}
+
+void	Message::setSrc(std::string str)
+{
+	this->_src = str;
+}
+
+void	Message::setNick(std::string str)
+{
+	this->_nick = str;
+}
+
+void	Message::setUser(std::string str)
+{
+	this->_user = str;
+}
+
+void	Message::setHostname(std::string str)
+{
+	this->_hostname = str;
+}
+
+void	Message::setVerb(std::string str)
+{
+	this->_verb = str;
+}
+
+// ------- MEMBER FONCTIONS -------
+
+void	Message::splitData(std::string tmp)
+{
+	std::istringstream	iss(tmp);
+	std::string token;
+
+	while (iss >> token)
+		this->_parameters.push_back(token);
+}
+
+void	Message::srcSplit(void)
+{
+	std::istringstream	buffer(this->_src);
+
+	std::getline(buffer, this->_nick, '!');
+	std::getline(buffer, this->_user, '@');
+	std::getline(buffer, this->_hostname, '\0');
+}
```

Comment:
What happens if there is no !@\0 ?
And not sure if its proper c++ to check for \0